VimeoNetworking/Sources/VimeoClient.swift

```diff
@@ -454,15 +460,39 @@ final public class VimeoClient
     
     // MARK: - Private error handling
     
-    private func handleError<ModelType: MappableResponse>(error: NSError, request: Request<ModelType>)
+    private func handleError<ModelType: MappableResponse>(error: NSError, request: Request<ModelType>, task: NSURLSessionDataTask? = nil)
     {
         if error.isServiceUnavailableError
         {
             Notification.ClientDidReceiveServiceUnavailableError.post(object: nil)
         }
         else if error.isInvalidTokenError
         {
-            Notification.ClientDidReceiveInvalidTokenError.post(object: nil)
+            Notification.ClientDidReceiveInvalidTokenError.post(object: self.token(fromTask: task))
+        }
+    }
+    
+    private func token(fromTask task: NSURLSessionDataTask?) -> String?
+    {
+        guard let task = task else
```

Comment:
this code could be cleaned up a bit.  The guard statement, the if statement, and the next guard statement can all be combined into a single guard statement.  

guard let bearerHeader = task.originalRequest?.allHTTPHeaderFields?[Constants.AuthorizationHeader],  
range = bearerHeader.rangeOfString(Constants.BearerQuery) else
{
    return nil
}